### PR TITLE
Fixes some bad assert statements and adds fuzzy floating point comparison 2

### PR DESCRIPTION
**Pull request by daw3rd**
## Why are these changes needed?

1. Need to fix asserts that were not being tested properly in test_support abstract_test.py
2. Adds support for floating point values to be within 1% of each other in parquet in anticipation of model scores being different values on different architectures.
3. Required updating some of the binary transform test files since we were not correctly comparing sizes until now.

### CODE REVIEW

File: data-processing-lib/python/src/data_processing/test_support/abstract_test.py [hunk 1]
```diff
@@ -96,7 +99,46 @@ def validate_expected_tables(table_list: list[pa.Table], expected_table_list: li
...
+        of each other as defined by global _allowed_float_percent_diff.
+        We assume the schema has already been compared and is equivalent.
+        Args:
+            table_index: index of tables that is the source of the rows.
+            row_index:
+            test_row:
+            expected_row:
+        """
+        assert test_row.num_rows == 1, "Invalid usage.  Expected test table with 1 row"
+        assert expected_row.num_rows == 1, "Invalid usage.  Expected expected table with 1 row"
```
Comment (blublinsky) [inline data-processing-lib/python/src/data_processing/test_support/abstract_test.py:118]: What exactly are we asserting here? That the table has more than 1 row? Why is this required?

File: data-processing-lib/python/src/data_processing/test_support/abstract_test.py [hunk 2]
```diff
@@ -96,7 +99,46 @@ def validate_expected_tables(table_list: list[pa.Table], expected_table_list: li
...
+        if test_row != expected_row:
+            # Else look for floating point values that might differ within the allowance
+            msg = f"Row {row_index} of table {table_index} are not equal\n\tTransformed: {test_row}\n\tExpected   : {expected_row}"
+            test_columns = test_row
+            expected_columns = test_row
```
Comment (blublinsky) [inline data-processing-lib/python/src/data_processing/test_support/abstract_test.py]: Why these extra variables? did you mean:
test_columns = test_row.schema.columns
expected_columns = expected_row.schema.columns

File: data-processing-lib/python/src/data_processing/test_support/abstract_test.py (hunk 1 above)
Comment (daw3rd) [reply data-processing-lib/python/src/data_processing/test_support/abstract_test.py:118]: only to be sure that the caller (us) is using this method correctly.

Comment (daw3rd) [reply data-processing-lib/python/src/data_processing/test_support/abstract_test.py:118]: and the code that follows expects a single row

File: data-processing-lib/python/src/data_processing/test_support/abstract_test.py (hunk 2 above)
Comment (daw3rd) [reply data-processing-lib/python/src/data_processing/test_support/abstract_test.py]: actually, there is a bug here.  expected_columns = test_row.  Will remove and fix bug.

File: data-processing-lib/python/src/data_processing/test_support/abstract_test.py [hunk 3]
```diff
@@ -96,7 +99,45 @@ def validate_expected_tables(table_list: list[pa.Table], expected_table_list: li
...
+        We assume the schema has already been compared and is equivalent.
+        Args:
+            table_index: index of tables that is the source of the rows.
+            row_index:
+            test_row:
+            expected_row:
+        """
+        assert test_row.num_rows == 1, "Invalid usage.  Expected test table with 1 row"
+        assert expected_row.num_rows == 1, "Invalid usage.  Expected expected table with 1 row"
+        if test_row != expected_row:
```
Comment (blublinsky) [inline data-processing-lib/python/src/data_processing/test_support/abstract_test.py:119]: This compares 2 tables, not the amount of columns in 2 tables. see https://arrow.apache.org/docs/python/generated/pyarrow.Schema.html#pyarrow.Schema.names to get column names

File: data-processing-lib/python/src/data_processing/test_support/abstract_test.py [hunk 4]
```diff
@@ -96,7 +99,45 @@ def validate_expected_tables(table_list: list[pa.Table], expected_table_list: li
...
+            # Else look for floating point values that might differ within the allowance
+            msg = f"Row {row_index} of table {table_index} are not equal\n\tTransformed: {test_row}\n\tExpected   : {expected_row}"
+            assert len(test_row) == len(expected_row), msg
```
Comment (blublinsky) [inline data-processing-lib/python/src/data_processing/test_support/abstract_test.py]: len(table) returns number of rows, not number of columns. I think you want to compare number of columns

File: data-processing-lib/python/src/data_processing/test_support/abstract_test.py (hunk 3 above)
Comment (daw3rd) [reply data-processing-lib/python/src/data_processing/test_support/abstract_test.py:119]: 1.  we've already verified the schema is equivalent 
2. what i think we want here is a full table equality check, which was what we were doing originally I believe. see https://github.com/IBM/data-prep-kit/blob/3ee26c500f55abdd1681ee5586bc4e8ea73d83a9/data-processing-lib/python/src/data_processing/test_support/abstract_test.py#L99 in 0.2.0

File: data-processing-lib/python/src/data_processing/test_support/abstract_test.py (hunk 4 above)
Comment (daw3rd) [reply data-processing-lib/python/src/data_processing/test_support/abstract_test.py]: nice.  done.

**Review by blublinsky — APPROVED** [review]
